### ISSUE TITLE
Asking for Feedback on Mesos Plugin 2.0

### ISSUE THREAD

**Issue by jeschkies** [jenkinsci/mesos-plugin#343]
Marathon/Mesosphere engineer here 👋 

As you might know Mesosphere has a fork of this plugin on [mesosphere/mesos-plugin](https://github.com/mesosphere/mesos-plugin/). It is used to support Jenkins on [DC/OS](https://dcos.io/). Unfortunately the code diverged over time. Furthermore user feature requests and load tests pushed us to rethink the architecture of the plugin. For instances we want to support suppress/revive or be non-blocking to scale up. Thus we decided to rewrite the plugin based on [USI](https://github.com/mesosphere/usi), a new second level scheduler for Mesos built with the learning from Marathon. You can find a working prototype in our branch [mesosphere/mesos-plugin#master](https://github.com/mesosphere/mesos-plugin/tree/master).

We would love to make this new plugin the 2.0 of the Mesos Plugin for Jenkins to avoid confusion among users, give back to the community but also receive feedback. That's why I'm reaching out to you to understand what we should do so that the plugin can be merged back upstream.

@Vlatombe, you are the main contributor so your feedback is very important to us. What do you and the others think? I'm happy to give more information.

**Comment by turb** [comment]
Hello @jeschkies, we woud be happy to test this, but I have a few question:

- is this branch testeable as is? (your link to the branch is dead, and I cannot find it)
- does USI needs some installation, or is it bundled?
- does it need a minimum Mesos version?
- does it support configuration migration?
- is it possible to rollback quickly?

**Comment by jeschkies** [comment]
@turb 

> is this branch testeable as is? (your link to the branch is dead, and I cannot find it)
We merged the feature branch into `master`. I relinked it. And yes, it's testable.

> does USI needs some installation, or is it bundled?
USI is a simple JVM library. We did not release it to Maven and use a source dependency. Feel free to check it out at https://github.com/mesosphere/usi.

> does it need a minimum Mesos version?
Mesos must support the v1 API which was introduced quite a while ago. However, I expect to have 1.4 or 1.5 as the minimal version.

> does it support configuration migration?
I hope. We are investigating two options.

> is it possible to rollback quickly?
The configuration will change quite a bit so I'd say no. One should back up the original config before an update. We might enable the installation of 2.0 alongside 0.x.
